src/FlaUI.WebDriver/Controllers/WindowController.cs

```diff
@@ -141,9 +141,7 @@ private IEnumerable<string> GetWindowHandles(Session session)
                 return Enumerable.Empty<string>();
             }
 
-            // GetAllTopLevelWindows sometimes times out, so we return only the main window and modal windows
-            // https://github.com/FlaUI/FlaUI/issues/596
-            var knownWindows = mainWindow.ModalWindows.Prepend(mainWindow)
+            var knownWindows = session.App.GetAllTopLevelWindows(session.Automation)
```

Comment:
The code at line 138-141 may also be removed

Reply:
Done.